File: 2571-find-the-pivot-integer/find-the-pivot-integer.py
class Solution:
    def pivotInteger(self, n: int) -> int:
        '''

        from equating the sum of 1...x element == sum of x..n elements
        we get  x = root(n*(n+1)//2


        
        '''
        tot = n*(n+1)//2
        for i in range(1,n+1):
            
            if i == tot**(0.5):

                return i
        return -1

File: 2571-find-the-pivot-integer/find-the-pivot-integer.java
public class Solution {
    public int pivotInteger(int n) {
        // Calculate the total sum of the sequence 1...n
        int tot = n * (n + 1) / 2;

        // Iterate over the sequence 1...n
        for (int i = 1; i <= n; i++) {
            // Check if the current element is equal to the square root of tot
            if (i == Math.sqrt(tot)) {
                return i; // If true, return the element
            }
        }
        return -1; // If no such element found, return -1
    }
}